src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -100,13 +98,14 @@ class NexusPublishPlugin : Plugin<Project> {
         }
         rootProject.tasks.withType(AbstractTransitionNexusStagingRepositoryTask::class.java).configureEach {
             transitionCheckOptions.convention(extension.transitionCheckOptions)
-            stagingRepositoryId.convention(registry.map { it[repository.get().name].stagingRepositoryId })
+            usesService(registryService)
+            stagingRepositoryId.convention(registryService.map { it.registry[repository.get().name].stagingRepositoryId })
         }
         extension.repositories.all {
             username.convention(rootProject.provider { rootProject.findProperty("${name}Username") as? String })
             password.convention(rootProject.provider { rootProject.findProperty("${name}Password") as? String })
```

Comment:
E2E test execution queued for [8e96e5e](https://github.com/gradle-nexus/publish-plugin/pull/274/commits/8e96e5eb3c6cb1d3a00cc9d9c254443ced17d7c7): https://github.com/gradle-nexus/publish-plugin/actions/runs/6224526816 (a copy of the branch)